### DISCUSSION TITLE
API' Calling issue

### DISCUSSION THREAD

**Discussion by 99003662** [zephyrproject-rtos/zephyr#41181] [Q&A]
Hi ,
     i am new to zephyr rtos and am struct in a position where I can't figure out how api's will be called..
                                          For ex-  am referring flash.h driver file , it has functions like **flash_read, flash_write** etc fine. now am using one of api , i.e. **flash_write ()** in my code fine. so that function internally calling one more function named **write()** ,with same arguments. now my query is what is that **write()** api is doing exactly , where is that api definition , what are the steps to be taken care inside that api , where to find that api definition .

**Comment by de-nordic** [comment]
The first parameter of each flash API call is a device pointer, this pointer is used to identify a flash device that will be used. The write callback you mention comes from flash API interface: https://github.com/zephyrproject-rtos/zephyr/blob/main/include/drivers/flash.h#L130-L143.
Each flash device driver defines the API separately, for example look at this definition https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/flash/soc_flash_nrf.c#L264-L272.
